openeo/rest/connection.py

```diff
@@ -1385,6 +1385,56 @@ def load_stac(
             _log.warning(f"Failed to extract cube metadata from STAC URL {url}", exc_info=True)
         return cube
 
+    def load_stac_from_job(
+        self,
+        job: Union[BatchJob, str],
+        spatial_extent: Optional[Dict[str, float]] = None,
+        temporal_extent: Union[Sequence[InputDate], Parameter, str, None] = None,
+        bands: Optional[List[str]] = None,
+        properties: Optional[Dict[str, Union[str, PGNode, Callable]]] = None,
+    ) -> DataCube:
+        """
+        Wrapper for :py:meth:`load_stac` that loads the result of a previous job using the STAC collection of its results.
+
+        :param job: The batch job object that has successfully finished, or a job id. When a job id is provided,
+            a job object is created using self as connection.
```

Comment:
"using self as connection" won't make sense to the general user.

I would phrase it something like

```
:param job: a :py:class:`~openeo.rest.job.BatchJob` or job id pointing to a finished job. 
    Note that the  :py:class:`~openeo.rest.job.BatchJob` approach allows to point 
    to a batch job on a different back-end.
```